### DISCUSSION TITLE
Differences in Beeswarm vs Summary Plot

### DISCUSSION THREAD

**Discussion by manavka** [shap/shap#3168] [Q&A]
Hello everyone,

I have been experimenting with beeswarms and summary plots and ran into a curious inconsistency between the summary and beeswarm plots. Specifically, the order of the beeswarm and summary plots seem to be different and for some reason the left side x axis seems to be squished/nonexistant in the beeswarm. All of these inconsistencies have been found out using the same data and models for both. Has anyone else experienced this? Am I missing something? Any info would be great! I have attatched my code and outputs as follows. 

`
## Beeswarm
explainer = shap.TreeExplainer(model_tuned, X_train)
shap_values = explainer(X_train)

shap.plots.beeswarm(shap_values,show=False)
top_features = [
    t.get_text() for t in reversed(plt.gca().yaxis.get_majorticklabels()) # Saving off top features
]
print(top_features)
string_dict["bee_feats"] = top_features
plt.title("Beeswarm SHAP", fontsize = 14, fontweight = 'bold')
plt.savefig("./Images/BeeswarmSHAP.png",bbox_inches="tight", )
string_dict["bee_SHAP"]="./Images/BeeswarmSHAP.png"

## Summary
Xd=xgb.DMatrix(X_train, label=y_train)
explainer = shap.TreeExplainer(model_tuned)
shap_values = explainer.shap_values(Xd)#pass it features and target
shap.summary_plot(shap_values, X_train.columns,show=False)
plt.title("Summary SHAP", fontsize = 14, fontweight = 'bold')
plt.savefig("./Images/SHAP.png",bbox_inches="tight")

string_dict["SHAP"]="./Images/SHAP.png"
`
![beeswarm1](https://github.com/shap/shap/assets/71353846/1e52cd88-a2e1-4ca4-b2ea-5be4255c7e63)

![summaryplot](https://github.com/shap/shap/assets/71353846/4523363d-73db-4eef-81ae-c2db41d7469e)

**Comment by znacer** [comment]
Hello,
To my knowledge, these 2 plots are the same. Summary is a legacy function. 
Are the 2 plots computed from the same trained model ? Dis you used the same explainer ?
The difference could come from the fact that you are representing explanations of 2 different model.

**Comment by yudeng2022** [comment]
I am encountering the same issue. Same models, same data, I got two different results using beeswarm vs summary plot
